src/mender-update/daemon/states.cpp

```diff
@@ -106,8 +114,26 @@ void IdleState::OnEnter(Context &ctx, sm::EventPoster<StateEvent> &poster) {
 void SubmitInventoryState::DoSubmitInventory(Context &ctx, sm::EventPoster<StateEvent> &poster) {
 	log::Debug("Submitting inventory");
 
-	auto handler = [&ctx, &poster](error::Error err) {
+	auto handler = [this, &ctx, &poster](error::Error err) {
 		if (err != error::NoError) {
+#ifdef MENDER_USE_DBUS
+			// If we get a DBus reply error, we should retry the polling with a short interval
+			// to allow the DBus authentication to be retried.
+			if (err.code == error_condition(dbus::ReplyError, dbus::DBusErrorCategory)) {
+				log::Error("Authentication over DBus failed, retrying");
+				poll_timer_.Cancel();
+				poll_timer_.AsyncWait(chrono::seconds(5), [&poster](error::Error err) {
+					if (err != error::NoError) {
+						if (err.code != make_error_condition(errc::operation_canceled)) {
+							log::Error("DBus poll timer caused error: " + err.String());
+						}
+					} else {
+						poster.PostEvent(StateEvent::InventoryPollingTriggered);
+					}
+				});
+			}
+#endif
+			void(this);
```

Comment:
This should not be needed. There's a private member `poll_timer_` being used so `this` is required to be captured. Did you get a warning somewhere?

Reply:
Yeah, unused variable warning in the no-dbus test.

Reply:
I wonder if we should do `this->poll_timer_` instead then. But it's strange, I believe we have this in other places too.

Reply:
Nothing from the class is being used here if dbus is compiled out, so the error makes sense, right? It only throws the error if we compile out dbus

Reply:
Oh! Should the capture be different in such a case then? If not, please use `#else` and add a comment why `void(this)` is needed in that case.

Reply:
Without the dbus the capture of `this` isn't needed, but I didn't want to overcomplicate it, but perhaps I should do:

```
#ifdef MENDER_USE_DBUS
    auto handler = [this, &ctx, &poster](error::Error err) {
         ....
 #else
    auto handler = [&ctx, &poster](error::Error err) {
    ...
#endif
```